### PR TITLE
Merge TripleStoreKnowledgeBase and TripleStore

### PR DESCRIPTION
**Pull request by Jean-KOUAGOU**
I have implemented a simple merge of TripleStore and TripleStoreKnowledgeBase. Tests work locally. Please review and merge

### CODE REVIEW

File: tests/test_triplestore.py [hunk 1]
```diff
@@ -53,12 +52,16 @@ def test_remote_triplestore_dbpedia_tdl(self):
...
+            print(h)
             assert h
             assert DLSyntaxObjectRenderer().render(h)
             save_owl_class_expressions(h)
             sparql = Owl2SparqlConverter().as_query("?x", h)
             assert sparql
         else:
             pass
-        """
+
```
Comment (Demirrr) [inline tests/test_triplestore.py:63]: Remove
if __name__ == "__main__":
    test_triplestore = TestTriplestore()
    test_triplestore.test_remote_triplestore_dbpedia_tdl()
    
 No need these three lines.

File: tests/test_triplestore.py [hunk 2]
```diff
@@ -53,12 +52,16 @@ def test_remote_triplestore_dbpedia_tdl(self):
             typed_neg = set(map(OWLNamedIndividual, map(IRI.create, examples["negative_examples"])))
             lp = PosNegLPStandard(pos=typed_pos, neg=typed_neg)
             h = model.fit(learning_problem=lp).best_hypotheses()
+            print(h)
```
Comment (Demirrr) [inline tests/test_triplestore.py:55]: No need to print

File: tests/test_triplestore.py [hunk 3]
```diff
@@ -53,12 +52,16 @@ def test_remote_triplestore_dbpedia_tdl(self):
...
             assert h
             assert DLSyntaxObjectRenderer().render(h)
             save_owl_class_expressions(h)
             sparql = Owl2SparqlConverter().as_query("?x", h)
             assert sparql
         else:
             pass
```
Comment (Demirrr) [inline tests/test_triplestore.py:62]: pass is problematic. This is not a really test

File: ontolearn/triple_store.py [hunk 4]
```diff
@@ -981,21 +903,89 @@ def domain_of_double_data_properties(self, prop: OWLDataProperty):
...
-        else:
-            self.g = reasoner
-        self.ontology = self.g
-        self.reasoner = self.g
+    ontology: TripleStoreOntology
+    reasoner: TripleStoreReasoner
 
+    def __init__(self, url: str=None):
+        assert url is not None, "url must be string"
+        self.g = TripleStoreReasonerOntology(url=url)
```
Comment (Demirrr) [inline ontolearn/triple_store.py:915]: This is incorrect. We cannot have TripleStoreOntology, TripleStoreReasoner, and TripleStoreReasonerOntology, can we ?

**Review by Demirrr — CHANGES_REQUESTED** [review]
You didn't really merge TripleStoreKnowledgeBase and TripleStore but rather put already defined class of TripleStoreKnowledgeBase into TripleStore. What is the benefits of this ?

**Comment by alkidbaci** [comment]
I noticed that `TripleStore` class is now inheriting from `KnowledgeBase` which is against the main reason we have `TripleStore` class in the first place. Becase of that we have to set the reasoner and the ontology of that class, and thats why we have `TripleStoreOntology`, `TripleStoreReasoner` and `TripleStoreReasonerOntology` at the same class.
It's kind of the right direction but not quite the right way. 

TDL and Drill works. Drill tried only on family dataset.

These changes can be considered temporary but I suggest we have a more long lasting solution for this problem.

**Pull request closed without merging**